Stats Revamp: Fix Insights cards not adapting to large device text size

staskus opened the pull request:
Fixes #19017

## Description

Labels in Stats cards truncate or do not fit well when font size increases. 

## Solution

Go card by card and find the most appropriate solution: a) Go multiline if possible b) Limit the expansion of the label c) Allow the label to truncate

### Views & Visitors
- Allow cell title to go multiline
- Assign minimum amount of pace for title and details (weeks) button so they wouldn't shrink indefinitely using width constraints
- Allow date labels (this week / previous week) as well as views labels to shrink if they don't fit anymore
- Limit the size of labels in x and y axis

### Comments/Likes/Followers Total
- Allow count number to increase - support dynamic type
- Allow guide text to increase/decrease - support dynamic type for attributed string

### Most Popular Time
- Allow labels to go multiline 

### Latest Post Summary
- Allow column titles (View &  Likes & Comments) to shrink. They are usually one-word labels so going multiline is not a good option.

### All-Time / This Year / Today
- Allow labels to go multiline

### Posting Activity
- Allow month labels Aug/Sep/Oct increase to an extent
- Allow legend (fewer posts & more posts) to go multiline. Otherwise, they don't fit at all for smaller screen sizes, even without increased font size

## Testing instructions

1. Enable statsRevamp feature flag.
2. Open Stats.
3. Click the Settings icon and add all the cards.
4. Increase & Decrease dynamic font size.  
5. Notice the most important information - titles and numbers increasing and fitting the card when dynamic font size increases.

## Regression Notes

1. Potential unintended areas of impact

- Breaking UI for Stats with StatsRevampV2 disabled
- Breaking newly created UI

6. What I did to test those areas of impact (or what existing automated tests I relied on)

Manual testing on devices with different screen sizes and different dynamic type

7. What automated tests I added (or what prevented me from doing so)

None

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Images & Videos

### Before

<img src="https://user-images.githubusercontent.com/4062343/193522213-a0ea47a5-3c09-4f72-b470-9bd1a45efe6b.png" alt="drawing" width="300"/>
<img src="https://user-images.githubusercontent.com/4062343/193522220-13bcc3c4-e4e4-42ec-80a0-628298df2f64.png" alt="drawing" width="300"/>
<img src="https://user-images.githubusercontent.com/4062343/193522223-996b9286-e862-498f-8512-85562dd1adad.png" alt="drawing" width="300"/>
<img src="https://user-images.githubusercontent.com/4062343/193522226-5efe49f8-8998-4882-97aa-48e669ee71e5.png" alt="drawing" width="300"/>
<img src="https://user-images.githubusercontent.com/4062343/193522228-94efe36a-4c6b-4198-815a-96dd0e3d2c0e.png" alt="drawing" width="300"/>

### After

<img src="https://user-images.githubusercontent.com/4062343/193522422-5cf234bf-951b-4758-80de-305ffe2e34d2.png" alt="drawing" width="300"/>
<img src="https://user-images.githubusercontent.com/4062343/193522424-455d3e53-aecd-419c-bee2-a6f85afe02f4.png" alt="drawing" width="300"/>
<img src="https://user-images.githubusercontent.com/4062343/193522430-e4ed84d8-960d-46cc-b695-2fdb43d7e0fc.png" alt="drawing" width="300"/>
<img src="https://user-images.githubusercontent.com/4062343/193522433-4bc4294d-3d21-428e-bc8e-2cf0020816b5.png" alt="drawing" width="300"/>
<img src="https://user-images.githubusercontent.com/4062343/193571016-772eeea8-58cb-46c5-9e72-87a532e89937.png" alt="drawing" width="300"/>
<img src="https://user-images.githubusercontent.com/4062343/193522428-6de19571-0f5d-4dec-bfd0-78840285cee7.png" alt="drawing" width="300"/>

staskus committed:
Stats cell heading label and details button dynamic type support

- To stop heading label or button from truncating when dynamic type increases set "adjustsFontSizeToFitWidth = true" to adjust the size of the text when it doesn't fit anymore
- Allow heading label to go multiline to be able to expand as much as possible
- Give heading label and details button maximum width constraint, so they wouldn't cause each other to shrink when expanding

staskus committed:
Don't truncate this week / previous week labels in Views & Visitors stats cell

- Use "adjustsFontSizeToFitWidth" to limit expansion of labels
- Set width constraints for previous and latest labels to allow expand labels same amount

staskus committed:
Limit the size of StatsLineChart x and y axis labels

staskus committed:
Expand guide label attributed text

- Make guideAttributes computed variable to it would be rebuilt every time traitCollectionDidChange gets called

staskus committed:
Increase countLabel font size

staskus committed:
Allow heading label to take all space if details button is not visible

staskus committed:
Do not truncate labels in MostPopular cell by allowing multiline labels

staskus committed:
Do not truncate labels in LatestPostSummary

staskus committed:
Don't truncate in AllTime/ThisYear cards

- Make title multiline so the naming would fit
- Allow ViewMore button to expand when text size increases

staskus committed:
Dynamic Type support in Posting Activity cell

- Support expansion of ViewMore button by making heightConstraint greaterThanOrEqual
- Do not truncate legend by making labels multiline and setting leading/trailing constraints
- Do not increase size of month description labels since month view does not increase as well

staskus committed:
Merge branch 'trunk' into fix/19017-stats-revamp-insights-cards-not-adapting-to-large-device-text-size

staskus committed:
Fix PostingActivity month label clipping when height changes

sla8c commented on WordPress/Classes/ViewRelated/Stats/Insights/Two Column Stats/TwoColumnCell.swift:
```diff
@@ -85,7 +85,7 @@ private extension TwoColumnCell {
         let showViewMore = !dataRows.isEmpty && statSection == .insightsAnnualSiteStats
         viewMoreView.isHidden = !showViewMore
-        rowsStackViewBottomConstraint.constant = showViewMore ? viewMoreHeightConstraint.constant : 0
+        rowsStackViewBottomConstraint?.isActive = showViewMore
```
Since the viewMoreHeightConstraint has been removed, I believe you can remove the IBOutlet reference on line 14?

staskus replied:
Good observation, I'll check if it's still needed.

sla8c commented:
Overall this is looking good.  Just a couple of visual comments; when there is a longer string for the label, the wrap can be hard to read.  I wonder if 50% max width for the button can perhaps be reduced in size?

Examples with `Followers Total` and `Comments Total`

![image](https://user-images.githubusercontent.com/88816724/193806286-eb89b70c-d89d-483b-b651-b71234fe4839.png)

![image](https://user-images.githubusercontent.com/88816724/193806345-9cbac790-08ee-471e-a4aa-7866fde0bf36.png)

staskus commented:
Thanks for the comments!

> I wonder if 50% max width for the button can perhaps be reduced in size?

Valid point, since titles are likely to be larger than button action. I'll try to see how it looks!

staskus committed:
More spacing for heading label and less spacing for details button

- Do not fill 100% of the cell, allow spacing between heading and details button to exist

staskus committed:
Remove unused viewMoreHeightConstraint

staskus commented:
> Overall this is looking good. Just a couple of visual comments; when there is a longer string for the label, the wrap can be hard to read. I wonder if 50% max width for the button can perhaps be reduced in size?

I experimented with different width constraints as well as different approaches but it was hard to pin-point a sweet spot. Since different labels will have different lengths in different languages. I limited the width of the button and increased the width of the heading label. I also kept the spacing between the label and the button a bit bigger.

### Regular
<img src="https://user-images.githubusercontent.com/4062343/193904917-16fe17b6-c5d5-4600-8e46-831844d911ee.png" alt="drawing" width="200"/>

### Large
<img src="https://user-images.githubusercontent.com/4062343/193904588-3b978693-0fe8-47b7-aa6c-9f47b9745535.png" alt="drawing" width="200"/>

### Extra Extra Extra Large
<img src="https://user-images.githubusercontent.com/4062343/193904595-eca1b6a4-658e-4702-91eb-a2ee0255a47a.png" alt="drawing" width="200"/>

sla8c reviewed (approved):
LGTM
